apps/site/assets/ts/schedule/components/schedule-finder/UpcomingDepartures.tsx

```diff
@@ -114,24 +150,54 @@ const CrTableRow = ({
   );
 };
 
+const modeIsBus = (journey: EnhancedJourney): boolean =>
+  journey.route.type !== 2;
+
 const TableRow = ({
-  input,
+  state,
   journey
-}: TableRowProps): ReactElement<HTMLElement> | null => {
+}: {
+  state: State;
+  journey: EnhancedJourney;
+}): ReactElement<HTMLElement> | null => {
   const { realtime } = journey;
 
   if (realtime.prediction === null) return null;
 
-  const contentComponent =
-    journey.route.type !== 2
-      ? () => <BusTableRow journey={journey} />
-      : () => <CrTableRow journey={journey} />;
+  const contentComponent = modeIsBus(journey)
+    ? () => <BusTableRow journey={journey} />
+    : () => <CrTableRow journey={journey} />;
 
   return (
     <Accordion
-      input={getAdjustedInput(input, journey)}
+      state={state}
+      journey={journey}
+      contentComponent={contentComponent}
+    />
+  );
+};
+
+const Accordion = ({
```

Comment:
I'm seeing some more things I didn't follow completely at first that are confusing me. What's the need for copying this `Accordion` instead of continuing to use the shared version? That feels like a red flag.

And relatedly, I like the idea of moving `AccordionRow` to its own file given that it's used in multiple places, but can it be the entire `Accordion` component that is moved into its own file (with `AccordionRow` as an internal sub-component)? That feels like a better level of abstraction for what these components that use the Accordion should need to care about.

Reply:
Because of the changes in this PR, although they look the same, we have now two _different_ `Accordion`s (from a behavior standpoint): one that just displays the data (i.e. the one in Upcoming Departures) and one that fetches the data when it's being expanded (i.e. the one in the Daily Schedule).
That is why the piece that I extracted was just the (accordion) row, because some props change.
I can take another look and see how I could extract the complete accordion.

Reply:
Ok. Yeah, conceptually it seems like you could have an accordion component that didn't have any knowledge of the contents nested inside it (or concern with loading the data for those contents), but I understand that's not directly related to this work. If you think it might be helpful to try to walk through something like that together let me know, or if you think it would be better to consider that a follow-up item to think about later that seems fine too.

Reply:
Indeed I'd like to do some refactoring with rows and accordions but ideally in a separate ticket if that makes sense.

Reply:
Sounds like a plan.